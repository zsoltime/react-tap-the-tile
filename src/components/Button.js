import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/macro';

const StyledButton = styled.button.attrs(props => ({
  type: props.type ? props.type : 'button',
}))`
  background-color: ${props =>
    props.primary ? '#46b980' : '#363636'};
  border: 0;
  border-radius: 0.25rem;
  color: ${props => (props.primary ? 'inherit' : '#fff')};
  font-size: 1.5rem;
  max-width: 12rem;
  padding: 0.75rem 1rem;
  user-select: none;
  width: 100%;

  &:focus {
    outline: thin dotted;
  }
`;

const Button = ({ children, className, onClick, ...props }) => (
  <StyledButton className={className} onClick={onClick} {...props}>
    {children}
  </StyledButton>
);

Button.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired,
};

export default Button;
