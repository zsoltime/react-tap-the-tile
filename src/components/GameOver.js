import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components/macro';

import Button from './Button';
import Logo from '../icons/Logo';
import { ModalText, Title } from './Common.style';

const StyledLogo = styled(Logo)`
  margin-bottom: 1rem;
`;

const GameOver = ({ highScore, onStart, score }) => {
  return (
    <>
      <Title>Game Over</Title>
      <StyledLogo />
      <ModalText>
        <p>Oh noes! You tapped a white tile!</p>
        <p>
          <strong>Score: {score}</strong>
        </p>
        <p>
          <strong>High Score: {highScore}</strong>
        </p>
        <p>Don't give up and try again.</p>
      </ModalText>
      <Button onClick={onStart} primary>
        New Game
      </Button>
    </>
  );
};

GameOver.propTypes = {
  highScore: PropTypes.number.isRequired,
  onStart: PropTypes.func.isRequired,
  score: PropTypes.number.isRequired,
};

export default GameOver;
