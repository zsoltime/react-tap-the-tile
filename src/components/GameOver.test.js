import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import Button from './Button';
import GameOver from './GameOver';
import { ModalText } from './Common.style';

describe('<GameOver />', () => {
  it('displays score and high score correctly', () => {
    const highScore = 429;
    const score = 57;
    const wrapper = shallow(
      <GameOver
        highScore={highScore}
        onStart={f => f}
        score={score}
      />
    );

    expect(wrapper.find(ModalText).render()).toMatchSnapshot();
  });

  it('calls onStart on click', () => {
    const onStart = jest.fn();
    const wrapper = shallow(
      <GameOver highScore={0} onStart={onStart} score={0} />
    );
    wrapper.find(Button).simulate('click');

    expect(onStart).toBeCalledTimes(1);
  });
});
