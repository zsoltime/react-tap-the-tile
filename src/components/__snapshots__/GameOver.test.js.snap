// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameOver /> displays score and high score correctly 1`] = `
.c0 {
  font-size: 1.25rem;
  font-weight: 300;
  line-height: 1.25;
  margin-bottom: 1rem;
  text-align: center;
}

.c0 p {
  margin: 0 0 1rem;
}

<div
  class="c0"
>
  <p>
    Oh noes! You tapped a white tile!
  </p>
  <p>
    <strong>
      Score: 57
    </strong>
  </p>
  <p>
    <strong>
      High Score: 429
    </strong>
  </p>
  <p>
    Don't give up and try again.
  </p>
</div>
`;
